File: CS/ViewModels/ViewModelBase.cs
using System.ComponentModel;

namespace DXRibbonNotepad.ViewModels {
    public class ViewModelBase : INotifyPropertyChanged {
        protected virtual void OnPropertyChanged(string propertyName) {
            if(PropertyChanged != null)
                PropertyChanged(this, new PropertyChangedEventArgs(propertyName));
        }
        public event PropertyChangedEventHandler PropertyChanged;
    }
}

File: VB/ViewModels/ViewModelBase.vb
Imports System.ComponentModel

Namespace DXRibbonNotepad.ViewModels

    Public Class ViewModelBase
        Implements INotifyPropertyChanged

        Protected Overridable Sub OnPropertyChanged(ByVal propertyName As String)
            RaiseEvent PropertyChanged(Me, New PropertyChangedEventArgs(propertyName))
        End Sub

        Public Event PropertyChanged As PropertyChangedEventHandler Implements INotifyPropertyChanged.PropertyChanged
    End Class
End Namespace
